core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -93,6 +96,46 @@
             ApplicationProtocolNames.HTTP_2,
             ApplicationProtocolNames.HTTP_1_1);
 
+    /*
+     * Validate {@code sslContext} is configured properly. If {@code sslContext} is configured as client
+     * context, or key store password is not given to key store when {@code sslContext} is created using key
+     * manager factory, the validation will fail and an exception will be raised.
+     */
+    private static SslContext validateSslContext(SslContext sslContext) throws SSLException {
+        if (!sslContext.isServer()) {
+            throw new IllegalArgumentException("sslContext: " + sslContext + " (expected: server context)");
+        }
+
+        SSLEngine serverEngine = null;
+        SSLEngine clientEngine = null;
+
+        try {
+            serverEngine = sslContext.newEngine(ByteBufAllocator.DEFAULT);
+            serverEngine.setUseClientMode(false);
+            serverEngine.setNeedClientAuth(false);
+
+            final SslContext sslContextClient =
+                    buildSslContext(SslContextBuilder::forClient, sslContextBuilder -> {});
+            clientEngine = sslContextClient.newEngine(ByteBufAllocator.DEFAULT);
+            clientEngine.setUseClientMode(true);
+
+            final ByteBuffer appBuf = ByteBuffer.allocate(clientEngine.getSession().getApplicationBufferSize());
+            final ByteBuffer packetBuf = ByteBuffer.allocate(clientEngine.getSession().getPacketBufferSize());
+
+            clientEngine.wrap(appBuf, packetBuf);
+            appBuf.clear();
+            packetBuf.flip();
+            serverEngine.unwrap(packetBuf, appBuf);
+        } catch (SSLException e) {
+            throw new SSLException("failed to validate SSL/TLS configuration: " + e.getMessage());
```

Comment:
But don't we need to set the cause of the `SSLException`? `, e`

Reply:
Thanks! I fixed it.
I just understood what
>  I don't think we need to repeat the exception type.  

means. :sob: